### ISSUE TITLE
Python indentation not functioning properly anymore

### ISSUE THREAD

**Issue by dvchoudh** [nvim-treesitter/nvim-treesitter#2499]
### Describe the bug

After some commit (I can't really pin down which one), the python indentation has completely stopped functioning for me. It used to do fairly well, until this commit. Now when I type this for example: `for i in range(10):` and hit enter the cursor only goes to the next line; it's supposed to indent 4 spaces (or whatever it's set to in the nvim config) which it doesn't. Just to confirm, I have removed the python treesitter parser and checked that it works perfectly without it.

### To Reproduce

1. Install the latest build of neovim nightly
2. Install the nvim-treesitter plugin
3. Install the python treesitter parser using `:TSInstall python`
4. Enable treesitter indentation in the nvim-treesitter setup function in your neovim config
5. Open any existing or new python file, type any valid python code and see if the indentation works

### Expected behavior

Open a python file, type `for i in range(10):` for example and when the user hits enter on their keyboard it goes to the next line and indents by 4 spaces. I'm currently having to do it myself, by hitting `<TAB>` when in the next line.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - WARNING: `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
  - OK: `node` found v16.14.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.2.1 20211203 (Red Hat 11.2.1-7)
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - json           ✓ ✓ ✓ ✓ . 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - toml           ✓ ✓ ✓ ✓ ✓ 
  - markdown       ✓ . ✓ . ✓ 
  - cpp            ✓ ✓ ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.7.0-dev+1041-g07c97fa02
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/devc/Repositories/neovim_development/neovim/build/config -I/home/devc/Repositories/neovim_development/neovim/src -I/home/devc/Repositories/neovim_development/neovim/.deps/usr/include -I/usr/include -I/home/devc/Repositories/neovim_development/neovim/build/src/nvim/auto -I/home/devc/Repositories/neovim_development/neovim/build/include
Compiled by devc@fedora

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/home/devc/.local/neovim/share/nvim"

Run :checkhealth for more info
```

**Comment by theHamsta** [comment]
I think this is the same issue as with #2497 and Lua earlier where a `")" @indent_end` might help to end the aligned indent

**Comment by dchuri** [comment]
Oh okay, should I close this issue then?

**Comment by theHamsta** [comment]
@dchuri you can try to apply a similar fixed like: https://github.com/nvim-treesitter/nvim-treesitter/pull/2505#discussion_r802876518 so making `[")"  "]" "}"] @indent_end`

**Comment by dchuri** [comment]
> @dchuri you can try to apply a similar fixed like: [#2505 (comment)](https://github.com/nvim-treesitter/nvim-treesitter/pull/2505#discussion_r802876518) so making `[")" "]" "}"] @indent_end`

I just tried that out, but it seems like it didn't fix it. I've pushed the changes to a separate branch, can you please review if I've made any mistakes or not? Thanks! https://github.com/dchuri/nvim-treesitter/commit/2bc3be6ecde404d7d2bc3103bfbb6d852e5a2862

**Comment by desilinguist** [comment]
Can confirm that I am seeing the same issue and the fix doesn't seem to work for me either.

**Comment by a1gx** [comment]
I had the same problem

**Comment by esn89** [comment]
Having same problem

**Comment by ChenZhongPu** [comment]
Same problem.

**Comment by dpetka2001** [comment]
> @dchuri you can try to apply a similar fixed like: [#2505 (comment)](https://github.com/nvim-treesitter/nvim-treesitter/pull/2505#discussion_r802876518) so making `[")" "]" "}"] @indent_end`

I think this fix is not corellated to the problem described in this issue. The fix is for ending indentation after `)`, so that the next line will not be indented as mentioned in [#2505](https://github.com/nvim-treesitter/nvim-treesitter/pull/2505). The problem here is that after a function definition or a `for` loop, which both end with `:`, there is no indentation in the next line as there should.

**Comment by dvchoudh** [comment]
Closing this as #4524 has vastly improved treesitter indentation for python. :)
